x-pack/platform/plugins/shared/alerting/server/application/backfill/transforms/transform_backfill_param_to_ad_hoc_run.ts

```diff
@@ -19,16 +19,18 @@ export const transformBackfillParamToAdHocRun = (
   actions: DenormalizedAction[],
   spaceId: string
 ): AdHocRunSO => {
-  const schedule = calculateSchedule(param.start, rule.schedule.interval, param.end);
+  const schedule = calculateSchedule(rule.schedule.interval, param.ranges);
   const shouldRunActions = param.runActions !== undefined ? param.runActions : true;
+  const start = param.ranges[0].start;
+  const end = param.ranges[param.ranges.length - 1].end;
```

Comment:
Nit: we refer to `param.ranges` several times here, it might be more legible if it were destructured:
```suggestion
  const { ranges } = param;
  const schedule = calculateSchedule(rule.schedule.interval, ranges);
  const shouldRunActions = param.runActions !== undefined ? param.runActions : true;
  const start = ranges[0].start;
  const end = ranges[ranges.length - 1].end;
```